src/components/vm/vmActions.tsx

```diff
@@ -44,164 +44,134 @@ import { RenameDialog } from './vmRenameDialog.jsx';
+            onShutOff: null,
+            error: {
+                text: msg,
+                detail: String(ex),
+            }
+        })
+    );
+}
+
+export async function vmStart(vm: VM) {
+    startOperationProgress(vm);
+    try {
+        await vmDomainMethod<void>(vm, 'Create', 'u', 0);
+    } catch (ex) {
+        setVmError(vm, cockpit.format(_("VM $0 failed to start"), vm.name), ex);
+    }
+}
 
-const onInstall = (vm: VM) => (
-    domainInstall({ vm }).catch(ex => {
+async function onInstall(vm: VM) {
+    try {
+        await domainInstall({ vm });
+    } catch (ex) {
         addNotification({
             text: cockpit.format(_("VM $0 failed to get installed"), vm.name),
-            detail: ex.message.split(/Traceback(.+)/)[0],
+            detail: String(ex).split(/Traceback(.+)/)[0],
             resourceId: vm.id,
         });
-    })
-);
+    }
+}
 
-const onReboot = (vm: VM) => (
-    domainReboot({ id: vm.id, connectionName: vm.connectionName }).catch(ex => {
-        store.dispatch(
-            updateVm({
-                connectionName: vm.connectionName,
-                name: vm.name,
-                error: {
-                    text: cockpit.format(_("VM $0 failed to reboot"), vm.name),
-                    detail: ex.message,
-                }
-            })
-        );
-    })
-);
+export async function vmReboot(vm: VM) {
+    try {
+        await vmDomainMethod<void>(vm, 'Reboot', 'u', 0);
+    } catch (ex) {
+        setVmError(vm, cockpit.format(_("VM $0 failed to reboot"), vm.name), ex);
+    }
+}
 
-const onForceReboot = (vm: VM) => (
-    domainForceReboot({ id: vm.id, connectionName: vm.connectionName }).catch(ex => {
-        store.dispatch(
-            updateVm({
-                connectionName: vm.connectionName,
-                name: vm.name,
-                error: {
-                    text: cockpit.format(_("VM $0 failed to force reboot"), vm.name),
-                    detail: ex.message,
-                }
-            })
-        );
-    })
-);
-
-const onShutdown = (vm: VM, setOperationInProgress: (val: boolean) => void) => (
-    domainShutdown({ id: vm.id, connectionName: vm.connectionName })
-            .then(() => !vm.persistent && cockpit.location.go(["vms"]))
-            .catch(ex => {
-                setOperationInProgress(false);
-                store.dispatch(
-                    updateVm({
-                        connectionName: vm.connectionName,
-                        name: vm.name,
-                        error: {
-                            text: cockpit.format(_("VM $0 failed to shutdown"), vm.name),
-                            detail: ex.message,
-                        }
-                    })
-                );
-            })
-);
+export async function vmForceReboot(vm: VM) {
+    try {
+        await vmDomainMethod<void>(vm, 'Reset', 'u', 0);
+    } catch (ex) {
+        setVmError(vm, cockpit.format(_("VM $0 failed to force reboot"), vm.name), ex);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2376-3aa4fc32-20251103-073120-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)